controllers/watcherapi_controller.go

```diff
@@ -18,42 +18,273 @@ package controllers
+			instance.Status.Conditions.Set(
+				instance.Status.Conditions.Mirror(condition.ReadyCondition))
+		}
+		err := helper.PatchInstance(ctx, instance)
+		if err != nil {
+			_err = err
+			return
+		}
+	}()
+
+	err = r.initStatus(instance)
+	if err != nil {
+		return ctrl.Result{}, nil
+	}
+
+	// If we're not deleting this and the service object doesn't have our finalizer, add it.
+	if instance.DeletionTimestamp.IsZero() && controllerutil.AddFinalizer(instance, helper.GetFinalizer()) || isNewInstance {
+		return ctrl.Result{}, nil
+	}
+
+	// Handle service delete
+	if !instance.DeletionTimestamp.IsZero() {
+		return r.reconcileDelete(ctx, instance, helper)
+	}
+
+	configVars := make(map[string]env.Setter)
+	// check for required OpenStack secret holding passwords for service/admin user and add hash to the vars map
+	Log.Info(fmt.Sprintf("[API] Get secret 1 '%s'", instance.Spec.Secret))
+	secretHash, result, secret, err := ensureSecret(
+		ctx,
+		types.NamespacedName{Namespace: instance.Namespace, Name: instance.Spec.Secret},
+		[]string{
+			instance.Spec.PasswordSelectors.Service,
+		},
+		helper.GetClient(),
+		&instance.Status.Conditions,
+		r.RequeueTimeout,
+	)
+	if (err != nil || result != ctrl.Result{}) {
+		return result, err
+	}
+
+	configVars[instance.Spec.Secret] = env.SetValue(secretHash)
+
+	// all our input checks out so report InputReady
+	instance.Status.Conditions.MarkTrue(condition.InputReadyCondition, condition.InputReadyMessage)
+
+	err = r.generateServiceConfigs(ctx, instance, secret, helper, &configVars)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	instance.Status.Conditions.MarkTrue(condition.ServiceConfigReadyCondition, condition.ServiceConfigReadyMessage)
+
+	// We reached the end of the Reconcile, update the Ready condition based on
+	// the sub conditions
+	if instance.Status.Conditions.AllSubConditionIsTrue() {
+		instance.Status.Conditions.MarkTrue(
+			condition.ReadyCondition, condition.ReadyMessage)
+	}
 
 	return ctrl.Result{}, nil
 }
 
+// generateServiceConfigs - create Secret which holds the service configuration
+// NOTE - jgilaber this function is WIP, currently implements a fraction of its
+// functionality and will be expanded of further iteration to actually generate
+// the service configs
+func (r *WatcherAPIReconciler) generateServiceConfigs(
+	ctx context.Context, instance *watcherv1beta1.WatcherAPI,
+	secret corev1.Secret,
+	helper *helper.Helper, envVars *map[string]env.Setter,
+) error {
+	Log := r.GetLogger(ctx)
+	Log.Info("generateServiceConfigs - reconciling")
+
+	db, err := mariadbv1.GetDatabaseByNameAndAccount(ctx, helper, watcher.DatabaseCRName, instance.Spec.DatabaseAccount, instance.Namespace)
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.ServiceConfigReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			condition.ServiceConfigReadyErrorMessage,
+			err.Error()))
+		return err
+	}
```

Comment:
so its possibel but unlikely that we will fail here if the db account does not exist at this point.